### PR TITLE
Layout: use override editor style API

### PR DESCRIPTION
**Pull request by ellatrix**
## What?

Uses the API introduced in #52888 for Layout.

### CODE REVIEW

File: packages/block-editor/src/hooks/layout.js [hunk 1]
```diff
@@ -404,26 +402,23 @@ export const withLayoutStyles = createHigherOrderComponent(
...
+			useDispatch( blockEditorStore )
+		);
+
+		useEffect( () => {
+			if ( ! css ) return;
+			setStyleOverride( id, { css } );
+			return () => {
+				deleteStyleOverride( id );
+			};
+		}, [ id, css, setStyleOverride, deleteStyleOverride ] );
```
Comment (youknowriad) [inline packages/block-editor/src/hooks/layout.js:415]: I wonder if it makes sense to have a `useStyleOverride` hook as this seems to be common right? At least an internal one.

File: packages/block-editor/src/hooks/layout.js [hunk 2]
```diff
@@ -404,26 +402,23 @@ export const withLayoutStyles = createHigherOrderComponent(
...
+
 		return (
-			<>
-				{ shouldRenderLayoutStyles &&
-					element &&
-					!! css &&
-					createPortal(
-						<LayoutStyle
```
Comment (youknowriad) [inline packages/block-editor/src/hooks/layout.js:413]: I don't understand what replaced this component?

Comment (youknowriad) [reply packages/block-editor/src/hooks/layout.js:413]: Oh looking at the component itself, it does nothing if it has a "css" prop, it just renders that css. Makes me think we should remove that css prop from the LayoutStyle component and just use `<style>` directly.

**Review by youknowriad — APPROVED** [review]
LGTM 👍

File: packages/block-editor/src/hooks/layout.js (hunk 1 above)
Comment (ellatrix) [reply packages/block-editor/src/hooks/layout.js:415]: Maybe!

**Pull request merged**